.github/workflows/ci-sbt.yml

```diff
@@ -43,6 +43,18 @@ jobs:
         with:
           cache: 'yarn'
 
+      - name: Add SSH key for accessing @guardian/private-infrastructure-config
+        run: |
+          mkdir -m 700 ~/.ssh
+          
+          ssh-keyscan -H github.com >> ~/.ssh/known_hosts
+
+          echo '${{ secrets.PRIVATE_INFRASTRUCTURE_CONFIG_DEPLOY_KEY }}' > ~/.ssh/id_ed25519
```

Comment:
I'd be tempted to follow the full instructions for this feature (including [the multiple repositories section](https://docs.github.com/en/developers/overview/managing-deploy-keys#using-multiple-repositories-on-one-server)) even though there's only one DeployKey being used here. This means giving this a more descriptive name to the key, adding some detail to `.ssh/config`, and specifying the identity when you clone the repository in the build itself.

This means being explicit about what the key is for, and explicitly using it when the repository is cloned. I think this will be much clearer to someone that arrives at this later, who might otherwise be surprised to see the clone working without any obvious authentication.

It also means we only have one approach to cloning private repositories, regardless how many need to be cloned. If we write it this way while this is simple then we have to know about two different approaches, and people won't be able to use it as a basis for more complex builds.

Let's chat about it though, no harm either way.

Reply:
Hey thanks for the link - really helpful. Definitely keen to go for a fuller solution (re-usable action probably) that supports more than one key etc.